### PR TITLE
Elastic Defend cannot run in emulation

### PR DESCRIPTION
**Pull request by intxgo**
## Issue

* https://github.com/elastic/elastic-agent/issues/6082

## What does this PR do?

Elastic Defend cannot run in emulation. The spec file should prevent installation in emulation giving a clear error message to the user.

## Why is it important?

When x86_64 Agent is installed by mistake on aarch64 Linux machine it won't be able to install x64_64 Elastic Defend. Furthermore `uninstall` command won't work.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Related issues

-

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: specs/endpoint-security.spec.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ inputs:
       - UPGRADE
     runtime:
       preventions:
+        - condition: ${runtime.arch} != ${runtime.native_arch}
+          message: "Elastic Defend cannot run in emulation"
```
Comment (blakerouse) [inline specs/endpoint-security.spec.yml:18]:
```suggestion
          message: "Elastic Defend cannot run on an emulated architecture"
```

**Review by blakerouse — APPROVED** [review]
Looks good.

Just a suggestion of the error that is logged.

**Comment by intxgo** [comment]
@pierrehilbert does the changelog look good? 1f01807e70ea3583ca4710296a104d6693ce56a4
I've used the tool for the first time.

**Comment by pierrehilbert** [comment]
We just need the fragment we don't need the other files in your PR

File: specs/endpoint-security.spec.yml [hunk 2]
```diff
@@ -14,6 +14,8 @@ inputs:
       - UPGRADE
     runtime:
       preventions:
+        - condition: ${runtime.arch} != ${runtime.native_arch}
```
Comment (cmacknz) [inline specs/endpoint-security.spec.yml:17]: Should this be done for MacOS as well, probably the platform most likely for this to occur?

https://github.com/elastic/elastic-agent/blob/434986ed9fa6fbed6257557a4c6773d519742f66/specs/endpoint-security.spec.yml#L51-L63

I swore I saw this change before and I did, but it was only for Windows https://github.com/elastic/elastic-agent/pull/4396/

Comment (intxgo) [reply specs/endpoint-security.spec.yml:17]: Perhaps you're right. For macOS we build a fat binary, containing executable for x86_64 and arm64, but I remember that the execution model follows the architecture of the parent process so x86_64 Agent will execute the fat binary as x64_64. I'll test it. Thanks

**Review by pierrehilbert — APPROVED** [review]

**Review by swiatekm — APPROVED** [review]

**Comment by intxgo** [comment]
I think the Unit Test failures are unrelated.

**Pull request merged**